overlord/snapstate/snapstate.go

```diff
@@ -1787,12 +1787,23 @@ func infoForUpdate(st *state.State, snapst *SnapState, name string, opts *Revisi
 // into the Autorefresh function.
 var AutoRefreshAssertions func(st *state.State, userID int) error
 
+// RefreshValidationSetAssertions allows to hook refreshing of validation sets
+// assertions into the Autorefresh function.
+var RefreshValidationSetAssertions func(st *state.State, userID int) error
+
 // AutoRefresh is the wrapper that will do a refresh of all the installed
 // snaps on the system. In addition to that it will also refresh important
 // assertions.
 func AutoRefresh(ctx context.Context, st *state.State) ([]string, []*state.TaskSet, error) {
 	userID := 0
 
+	if RefreshValidationSetAssertions != nil {
+		if err := RefreshValidationSetAssertions(st, userID); err != nil {
```

Comment:
instead of changing things here, I think the intention was always that AuthRefreshAssertions would do any necessary assertion refresh,  so the code that needs updating is assertstate.AutoRefreshAssertions